packages/search-ui-analytics-plugin/src/index.ts

```diff
@@ -1,16 +1,118 @@
-import type { Event } from "@elastic/search-ui";
+import {
+  Event,
+  BaseEvent,
+  ResultSelectedEvent,
+  SearchQueryEvent
+} from "@elastic/search-ui";
 
-export type EventType = "search" | "click" | "pageview";
+export type EventType = "search" | "search_click" | "page_view";
+
+export type DocumentAttribute = {
+  index: string;
+  id: string;
+};
+
+export type PageEventAttribute = {
+  referrer?: string;
+  url: string;
+  title?: string;
+};
+
+type SortAttribute = {
+  name: string;
+  direction?: "asc" | "desc";
+};
+
+export type ResultItemAttribute = {
+  document: DocumentAttribute;
+  page: {
+    url: string;
+  };
+};
+
+interface SearchEventAttribute {
+  search: {
+    query: string;
+    filters?: Record<string, string | string[]>;
+    search_application?: string;
+    page?: {
+      current: number;
+      size: number;
+    };
+    sort?: SortAttribute | SortAttribute[];
+    results?: {
+      items: ResultItemAttribute[];
+      total_results: number;
+    };
+  };
+}
+
+type SearchClickEventAttribute = SearchEventAttribute &
+  ({ page: PageEventAttribute } | { document: DocumentAttribute });
 
 export type AnalyticsClient = {
-  trackEvent: (eventType: EventType, payload: Record<string, any>) => void;
+  trackEvent: (
+    eventType: EventType,
+    payload: SearchEventAttribute | SearchClickEventAttribute
+  ) => void;
 };
 
 export interface AnalyticsPluginOptions {
   client?: AnalyticsClient;
 }
 
-export default function AnalyticsPlugin(options = { client: undefined }) {
+const mapEventToTrackerParams: Record<
+  ResultSelectedEvent["type"] | SearchQueryEvent["type"],
+  (
+    event: BaseEvent
+  ) =>
+    | ["search_click", SearchClickEventAttribute]
+    | ["search", SearchEventAttribute]
+> = {
+  ResultSelected: (event: ResultSelectedEvent) => [
+    "search_click",
+    {
+      document: { id: event.documentId, index: event.origin },
+      search: {
+        query: event.query,
+        results: {
+          items: [],
+          total_results: event.totalResults
+        },
+        search_application: "search-ui"
```

Comment:
exclude search_application. This is reserved for the name of the search application, not the client. This can added later on once we clarify the status of `search_application`